lib/web_ui/lib/src/engine/skwasm/skwasm_impl/renderer.dart

```diff
@@ -395,11 +396,13 @@ class SkwasmRenderer implements Renderer {
     if (contentType == null) {
       throw Exception('Could not determine content type of image at url $uri');
     }
-    return SkwasmImageDecoder(
+    final SkwasmImageDecoder decoder = SkwasmImageDecoder(
       contentType: contentType,
       dataSource: response.body as JSAny,
       debugSource: uri.toString(),
     );
+    await decoder.initialize();
+    return decoder;
```

Comment:
If `initialize` didn't swallow the returned future type here:

* https://github.com/flutter/engine/blob/6f9cc079605cc71c112a2c8c151fa923e8979e40/lib/web_ui/lib/src/engine/image_decoder.dart#L92-L94

(And returned the initialized `this` or something.

You could just `return decoder.initialize();`